pkg/image/api/types.go

```diff
@@ -60,6 +77,30 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
+	// Describes a type of stored blob.
+	Type string
+	// An opaque binary string which is an image's signature.
+	Content []byte
+	// A time of last server-side validation. IOW a time of the last update to ValidationResult.
+	LastValidation *unversioned.Time
+	// Result of validation performed at LastValidation. Filled by server.
+	ValidationResult SignatureValidationResult
```

Comment:
(I am not a native speaker, feel free to ignore me, but) it seems to me that `ValidationResult` only implies that the signature has been validated at the `LastValidation` time, with no claim on current validity. `SignatureState` seems more authoritative, suggesting up-to-date-state to me.